### PR TITLE
refactor: update commitlint and remove unnecessary changelog angular dependency

### LINKED ISSUE #776: reduce dependencies
### User story

For our current changelog flow we use conventional cimmits and the `conventional-changelog-angular@^7.0.0`package.
This has a dependency to `@angular-devkit/build-angular@^17`. 

### Acceptance criteria

- [ ] Check if it is even in use
- [ ] Get rid of angular as dependency by switching or re-creting the conventions

### Implementation details

_No response_

### PR DESCRIPTION
**Pull request by vmasek**
closes #776

### CODE REVIEW

File: package-lock.json
Comment (matejchalk) [inline package-lock.json:1]: Lock file is slightly out-of-sync, running `npm install` results in this change:

image

**Review by matejchalk — COMMENTED** [review]
The `conventional-changelog-angular` package is still installed via indirect dependencies. However, there is no `@angular-devkit/build-angular` installed.

image

It's not installed on `main` either, so I'm not sure what the point of this issue was :sweat_smile: 

image

But nice to have Commitlint updated and config converted to ESM.

File: package-lock.json
Comment (vmasek) [reply package-lock.json:1]: fixed

Comment (vmasek) [reply package-lock.json:1]: I'll rebase main and investigate

**Review by matejchalk — APPROVED** [review]

**Pull request merged**